### PR TITLE
travis: Use beaver v0.2.0 to simplify the build and use specific DMD versions in the matrix

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
Test run of https://github.com/sociomantic-tsunami/beaver/pull/18.

### CODE REVIEW

**Comment by leandro-lucarella-sociomantic** [comment]
:green_heart: 

I still need to release beaver and those things, but it seems like a good start.

**Comment by leandro-lucarella-sociomantic** [comment]
@nemanja-boric-sociomantic have you seen this PR, it might clarify what `DMD_PKG` is for and then we can improve the README in beaver.

**Comment by nemanja-boric-sociomantic** [comment]
Yup, I haven't seen this one!

**Comment by leandro-lucarella-sociomantic** [comment]
Update with the latest full automation beaver PR and :green_heart:!

Still will need to release beaver before merging.

**Comment by nemanja-boric-sociomantic** [comment]
I'll help you in that noble quest.

**Comment by nemanja-boric-sociomantic** [comment]
Looks very nice :+1:

**Comment by nemanja-boric-sociomantic** [comment]
@leandro-lucarella-sociomantic please update submodule and let's get this rolling!

**Comment by leandro-lucarella-sociomantic** [comment]
Updated!

**Comment by nemanja-boric-sociomantic** [comment]
:tada:

File: .travis.yml [hunk 1]
```diff
@@ -19,14 +19,14 @@ env:
...
-        - DMD=dmd1 DIST=xenial F=production AFTER_SCRIPT=1
-        - DMD=dmd1 DIST=xenial F=devel
-        - DMD=dmd-transitional DIST=trusty F=production
-        - DMD=dmd-transitional DIST=trusty F=devel
-        - DMD=dmd-transitional DIST=xenial F=production
-        - DMD=dmd-transitional DIST=xenial F=devel
+        - DMD=1.080.0 DIST=trusty F=production
+        - DMD=1.080.0 DIST=trusty F=devel
+        - DMD=1.080.0 DIST=xenial F=production AFTER_SCRIPT=1
+        - DMD=1.080.0 DIST=xenial F=devel
```
Comment (nemanja-boric-sociomantic) [inline .travis.yml]: `v1.081.0` is now out as well.

**Comment by leandro-lucarella-sociomantic** [comment]
This looks a bit fishy to me: https://travis-ci.org/sociomantic-tsunami/ocean/jobs/265516912#L583, this it's the default versions installed, it might be OK, but if you add a matrix to test a newer version, please test that this version is really used. Maybe we should do some check in the build.

**Comment by leandro-lucarella-sociomantic** [comment]
Coverage is complaining about `src/ocean/math/random/engines/KissCmwc.d` :man_facepalming:

**Comment by nemanja-boric-sociomantic** [comment]
We could check by using `v1.081.0` here, since it's not default one.

**Comment by leandro-lucarella-sociomantic** [comment]
Added, and removed trusty support to avoid matrix explosion.

**Comment by nemanja-boric-sociomantic** [comment]
ok:

```
Unpacking dmd1 (1.081.0-xenial) over (1.080.0-xenial) ...
```

https://travis-ci.org/sociomantic-tsunami/ocean/jobs/265519594

**Comment by nemanja-boric-sociomantic** [comment]
btw:

```
W: --force-yes is deprecated, use one of the options starting with --allow instead.
```

**Comment by leandro-lucarella-sociomantic** [comment]
All good: https://travis-ci.org/sociomantic-tsunami/ocean/jobs/265519595#L583

I think it could go as it is?

**Comment by nemanja-boric-sociomantic** [comment]
Yes!

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Comment by leandro-lucarella-sociomantic** [comment]
> btw:
> 
> W: --force-yes is deprecated, use one of the options starting with --allow instead

https://github.com/sociomantic-tsunami/beaver/blob/19f588867f4cb31c3155dfcda4ebb193396229f8/bin/dlang/install#L23-L25

**Pull request merged**